src/RA_AchievementSet.cpp

```diff
@@ -480,85 +450,83 @@ BOOL AchievementSet::LoadFromFile(ra::GameID nGameID)
                     }
                     else if (isdigit(buffer[0]))
                     {
-                        Achievement& newAch = g_pLocalAchievements->AddAchievement();
-                        buffer[nCharsRead - 1] = '\0';	//	Turn that endline into an end-string
+                        auto& newAch{ g_pLocalAchievements->AddAchievement() };
+                        auto szCharsRead{ static_cast<std::size_t>(ra::to_unsigned(nCharsRead)) };
+                        buffer[szCharsRead-1U] = '\0';	//	Turn that endline into an end-string
 
-                        newAch.ParseLine(buffer);
+                        newAch.ParseLine(buffer.get());
                     }
                 }
             }
 
-            fclose(pFile);
             return TRUE;
         }
         else
         {
-            Document doc;
-            doc.ParseStream(FileStream(pFile));
-            if (!doc.HasParseError())
+            rapidjson::Document doc;
+            rapidjson::IStreamWrapper isw{ ifile };
+            doc.ParseStream(isw);
+
+            if (doc.HasParseError())
             {
-                //ASSERT( doc["Success"].GetBool() );
-                g_pCurrentGameData->ParseData(doc);
-                ra::GameID nGameID = g_pCurrentGameData->GetGameID();
+                ASSERT(!"Could not parse file?!");
+                return FALSE;
+            }
+
+            //ASSERT( doc["Success"].GetBool() );
+            g_pCurrentGameData->ParseData(doc);
+            auto nGameID{ g_pCurrentGameData->GetGameID() };
 
-                //	Rich Presence
-                _WriteBufferToFile(g_sHomeDir + RA_DIR_DATA + std::to_string(nGameID) + "-Rich.txt", g_pCurrentGameData->RichPresencePatch());
-                g_RichPresenceInterpreter.ParseFromString(g_pCurrentGameData->RichPresencePatch().c_str());
+            //	Rich Presence
+            std::ostringstream oss;
+            oss << g_sHomeDir << RA_DIR_DATA << nGameID << "-Rich.txt";
+            auto richFile{ oss.str() };
+            oss.str("");
+            _WriteBufferToFile(richFile, g_pCurrentGameData->RichPresencePatch());
+            g_RichPresenceInterpreter.ParseFromString(g_pCurrentGameData->RichPresencePatch().c_str());
 
-                const Value& AchievementsData = doc["Achievements"];
-                for (SizeType i = 0; i < AchievementsData.Size(); ++i)
+            const auto& AchievementsData{ doc["Achievements"] };
+            for (auto& achData : AchievementsData.GetArray())
+            {
+                //	Parse into correct boxes                   
+                if (auto nFlags{ achData["Flags"].GetUint() }; ((nFlags == 3U) and (m_nSetType == Core)))
```

Comment:
All the bitwise and logical operators technically are keywords since C99/C++11, but haven't been added to VC++ yet. These were from iso646.h (C95). They'll be removed it causes controversy, will limit parentheses.